### PR TITLE
fix: add max idle time logic

### PR DESCRIPTION
**Pull request by gxcsoccer**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

心跳超时时间和心跳间隔区分开，默认是两倍心跳间隔

### CODE REVIEW

File: lib/base.js [hunk 1]
```diff
@@ -406,9 +406,14 @@ class TCPBase extends Base {
...
 
     if (this.options.needHeartbeat) {
+      const interval = this.options.heartbeatInterval;
+      let maxIdleTime = this.options.maxIdleTime;
+      if (!maxIdleTime || maxIdleTime < interval * 2) {
+        maxIdleTime = interval * 2;
+      }
       this._heartbeatTimer = setInterval(() => {
         const duration = this._lastHeartbeatTime - this._lastReceiveDataTime;
-        if (this._lastReceiveDataTime && duration > this.options.heartbeatInterval) {
```
Comment (leoner) [inline lib/base.js:411]: 主要就是这里放大了一下时间是把.

Comment (gxcsoccer) [reply lib/base.js:411]: 对放大一倍

Comment (gxcsoccer) [reply lib/base.js:411]: 心跳周期和断连的最大时间其实是两个概念

File: test/index.test.js [hunk 2]
```diff
@@ -203,8 +202,8 @@ describe('test/index.test.js', () => {
     client.close();
 
     client.send(makeRequest(1), err => {
-      should.exist(err);
-      err.name.should.eql('SocketCloseError');
+      assert(err);
```
Comment (leoner) [inline test/index.test.js:205]: 这句有点多余.

**Comment by leoner** [comment]
+1

**Comment by gxcsoccer** [comment]
1.1.1

**Pull request merged**